int_scan.hpp

```diff
@@ -0,0 +1,11 @@
+#ifndef __INT_SCAN_HPP
+#define __INT_SCAN_HPP
+
+#define INT_SCANNED 0;
+#define OVERFLOW_ERROR -1;
+#define OUT_OF_BOUNDS_ERROR -2;
+#define EMPTY_NUM_EXPR_ERROR -3;
```

Comment:
Do you think I should keep these macros or there may be conflicts later in the implementation?

Reply:
As I've said, this is not the final form of the function. Just trying to make it somewhat simple and generic before moving on to a more specific usage.
In this more specific form, these won't be used, because the function won't return an error, but instead will print warnings. There are reasons for this that I will explain when we move to implement the function in its final form.